alf/layers.py

```diff
@@ -3467,3 +3467,24 @@ def _make_spec(spec):
                                          spec.minimum, spec.maximum)
 
     return map_structure(_make_spec, specs)
+
+
+class AMPWrapper(nn.Module):
+    """Wrap a layer to run in a given AMP context.
+
+    Args:
+        enabled: whether to enable AMP autocast
+        net: the wrapped network
+    """
+
+    def __init__(self, enabled: bool, net: nn.Module):
+        super().__init__()
+        self._net = net
+        self._enabled = enabled
+
+    def forward(self, input):
+        if torch.is_autocast_enabled() and not self._enabled:
+            input = alf.nest.map_structure(
+                lambda x: x.float() if x.dtype.is_floating_point else x, input)
```

Comment:
If I understand this correctly, the design is to use `AMPWrapper(false)` to force disabling AMP. If so, it would be easier to read if we provide a separate `DisableAMPWrapper` instead.

Reply:
We want to able to easily switch between "enable" and "disable" to compare performance. Using a variable for it is easier.

Reply:
Make sense. Thanks for the clarification!